### ISSUE TITLE
GPU support for run.c

### ISSUE THREAD

**Issue by MostWrong** [karpathy/llama2.c#141]
Someone should add cuda support to run.c, I'm thinking of trying it out on the weekend.

**Comment by Ea0011** [comment]
I think somoeone with a gpu can try to set this up using OpenMP #pragma omp target directive. It supports CUDA and Apple GPU of I recall correctly. At the same time is really simple and easy to use so it will not bloat the code and make it complex. Sadly, i have only an AMD GPU so can't dig into this.

A nice reference here https://enccs.github.io/openmp-gpu/target/

**Comment by kroggen** [comment]
Wow! Using OpenMP with GPU would be a very interesting thing.

> i have only an AMD GPU so can't dig into this

An RTX 6000 costs just USD 0.50 per hour on [Lambda Labs](https://lambdalabs.com/service/gpu-cloud)

**Comment by kroggen** [comment]
You guys can also check #159

**Comment by Ea0011** [comment]
@kroggen Yes, OpenMP is amazing in my books ;]. With a single line of directive you can more or less have a GPU port. It is maybe not the most optimal but it's definitely easy. I think I might still give it a go and see what happens.

**Comment by RahulSChand** [comment]
I have tried to do this using OpenMP here: #166 . Open to suggestions

**Comment by Ea0011** [comment]
Wow this was quick! I'll take a look with a fresh mind tomorrow.
